### DISCUSSION TITLE
Ability to add "optional" fields to weather forecasts

### DISCUSSION THREAD

**Discussion by jeeftor** [home-assistant/architecture#925] [Entity Models]
There are a couple proposals to have different fields (Cloud Cover, Wind Gusts, UV Index) built into the weather forecast models. 

I'm wondering if perhaps there is a method to have free_text / optional fields added to the forecast in the case that not all weather providers offer these specific forecast elements...

**Comment by Jc2k** [comment]
Another proposal (which iirc has buy in from core devs to some degree) was to add the ability to provide forecasts for any sensor.

**Reply by frenck** [reply]
That would be me, but that I will close. Times changed :)

**Comment by frenck** [comment]
> I'm wondering if perhaps there is a method to have free_text / optional fields added to the forecast in the case that not all weather providers offer these specific forecast elements...

We don't have to have random/unstructured data. In general, we have state attributes that may help you out. However, those are limited and generally, we are not to keen on adding attributes.

../Frenck

**Reply by jeeftor** [reply]
I'll close this then as it sounds like its a no dice...

(Just trying to figure out a way for people to not have to use custom rest sensors to get both AQI and UV Forecasts... :))
